lib/package.py

```diff
@@ -101,8 +101,31 @@ def __init__(self, name, force_rebuild=True):
             raise exception.PackageDescriptorError(
                 "Failed to find %s's YAML descriptor" % self.name)
 
+        # load package metadata YAML file
         self._load()
 
+        # get global config information which may override YAML
+        packages = CONF.get('build_packages').get('packages') or []
+        for package in packages:
+            if package.startswith(name):
+                if package == name:
+                    break
+                else:
+                    package_parts = package.split("#")
```

Comment:
I think adding a comment here describing the expected format helps.  For example:

```python
# Parse and set package source attributes from the command line.
# Fields are separated by `#` character and the expected format
# can be one of the following:
#
#     package_name#repo_url#branch_name#revision_id
#     package_name#repo_url#reference
#     package_name##reference
#     package_name

package_parts = package.split("#")
...
```